week3/homework/src/index.js

```diff
@@ -1,3 +1,85 @@
 'use strict';
 
-// TODO: Write the homework code in this file
+const express = require('express');
+const bodyParser = require('body-parser');
+const fs = require('fs');
+
+const app = express();
+
+app.use(bodyParser.json());
+
+app.get('/todo/:id', (request, response) => {
+  fs.readFile('./todos.json', 'utf8', (error, todo) => {
+    if (error) {
+      if (error.code === 'ENOENT') {
+        console.log('no data found');
+        response.status(404).send('');
```

Comment:
This line should be moved to outer the `if...else`, so you always respond to user.

    if (error) {
      if (error.code === 'ENOENT') {
        console.log('no data found');
      } else console.error(error);
      response.status(404).send('');
    } else {
      const newList = todo.split('\n');
      const id = request.params.id;
      response.json({ result: newList[id] });
    }

Reply:
Ok I will do ASAP.

Reply:
Thanks @eidosam I have corrected it.